packages/data-stores/src/plans/queries/use-site-plans.ts

```diff
@@ -17,38 +16,40 @@ interface Props {
 }
 
 /**
+ * Plans from `/sites/[siteId]/plans` endpoint, transformed into a map of planSlug => SitePlan
  * - Plans from `/sites/[siteId]/plans`, unlike `/plans`, are returned indexed by product_id, and do not include that in the plan's payload.
  * - UI works with product/plan slugs everywhere, so returned index is transformed to be keyed by product_slug
  */
-function useSitePlans( { siteId }: Props ) {
+function useSitePlans( { siteId }: Props ): UseQueryResult< SitePlansIndex > {
 	const queryKeys = useQueryKeysFactory();
 
-	return useQuery< PricedAPISitePlansIndex, Error, SitePlansIndex >( {
+	return useQuery( {
 		queryKey: queryKeys.sitePlans( siteId ),
-		queryFn: async () =>
-			await wpcomRequest( {
+		queryFn: async () => {
```

Comment:
From what I understand, the `select` function is only called if data exists ( [article]( https://tkdodo.eu/blog/react-query-data-transformations#1-in-the-queryfn ) ) . Wondering if we need to check that `data` is meaningful this new queryFn transformation before passing it to`Object.keys` 🤔

Reply:
Good question. I suppose we don't have to, though. 
The data is typed as `PricedAPISitePlansIndex`, so by the time it is passed to `Object.keys` it should be a properly formatted object. If there is any conversion error or `wpcomRequest` has failed, it looks like this part of the code will throw, which looks like the expected way of communicating `queryFn` error for react-query: https://tanstack.com/query/v3/docs/react/guides/query-functions. 

I don't have decent experience with react-query, so happy to be corrected :)